packages/react-scripts/scripts/eject.js

```diff
@@ -36,6 +37,31 @@ prompt(
     process.exit(1);
   }
 
+  // Make sure there are no dirty git status
+  const git = fs.existsSync(path.join(process.cwd(), '.git'));
```

Comment:
See https://github.com/facebookincubator/create-react-app/pull/1288/files#diff-0ae46a2383dedcac5f5e369f5ca5f169R16 for inspiration.